pkg/reconciler/pipelinerun/pipelinerun_expired.go

```diff
@@ -0,0 +1,145 @@
+package pipelinerun
+
+import (
+	"fmt"
+	"time"
+
+	apispipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/controller"
+)
+
+const ControllerName = "TTLExpiredController"
+
+func (tc *Reconciler) AddPipelineRun(obj interface{}) {
+	pr := obj.(*apispipeline.PipelineRun)
+	tc.Logger.Infof("Adding PipelineRun %s/%s if the PipelineRun has succeeded or failed and has a TTL set.", pr.Namespace, pr.Name)
+
+	if pr.DeletionTimestamp == nil && pipelineRunCleanup(pr) {
+		controller.NewImpl(tc, tc.Logger, ControllerName).Enqueue(pr)
+	}
+}
+
+func (tc *Reconciler) UpdatePipelineRun(old, cur interface{}) {
+	pr := cur.(*apispipeline.PipelineRun)
+	tc.Logger.Infof("Updating PipelineRun %s/%s if the PipelineRun has succeed or failed and has a TTL set.", pr.Namespace, pr.Name)
+
+	if pr.DeletionTimestamp == nil && pipelineRunCleanup(pr) {
+		controller.NewImpl(tc, tc.Logger, ControllerName).Enqueue(pr)
```

Comment:
I am confused about this.  `TTLExpiredController` is an alpha-level kubernetes controller, is that right?  And here we are creating an instance of that controller and enqueuing a pipelinerun for it.  But the kubernetes docs for the controller say that it only works for Jobs.  So what does this do exactly?